ecosystem/indexer-grpc/indexer-grpc-integration-test-transaction-generator/src/transaction_generator.rs

```diff
@@ -0,0 +1,273 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    capture_transactions::capture_transactions, move_executor::MoveExecutor,
+    DEFAULT_LOCAL_FAUCET_URL, DEFAULT_LOCAL_NODE_API_URL,
+};
+use anyhow::Context;
+use clap::Parser;
+use reqwest::Client;
+use std::{path::PathBuf, time::Duration};
+use tokio::{process::Child, time::sleep};
+
+const NODE_HEALTH_CHECK_COUNT: u8 = 60;
+const TEST_CASE_FOLDER_PREFIX: &str = "test_case_";
+const TEST_CASE_SETUP_MODULE_PREFIX: &str = "setup_";
+const TEST_CASE_STEP_PREFIX: &str = "step_";
+
+/// Args to start the transaction generator.
+#[derive(Debug, Parser)]
+pub struct TransactionGeneratorArgs {
+    /// The path to the test cases main folder.
+    #[clap(long)]
+    pub test_cases_folder: PathBuf,
+
+    /// The path of generated test cases.
+    #[clap(long)]
+    pub output_test_cases_folder: PathBuf,
+
+    /// The path of the aptos node binary.
+    /// This adds the ability to use a custom binary for the local node.
+    #[clap(long)]
+    pub aptos_node_binary: PathBuf,
+
+    /// The path of local node config file to override the default config.
+    #[clap(long)]
+    pub node_config: Option<PathBuf>,
+
+    /// The node version; this is to generate the test cases for a specific version.
+    /// If not provided, it will default to the `main` branch.
+    #[clap(long, default_value = "main")]
+    pub node_version: String,
```

Comment:
The comment implies we'd use this for determining which node to run, but we don't use it that way. So why do we need it? Feels like if we want to separate values based on the node version, we could just use `aptos-node --version` and get the git hash or something.